src/parser.rs

```diff
@@ -30,18 +39,23 @@ fn ascii_or_hyphen(chr: u8) -> bool {
     (chr >= 97 && chr <= 122)
 }
 
-/// parse a word
+/// Parse a word
 fn word(i: &[u8]) -> IResult<&[u8], String> {
     map_res!(i,
              take_while!(ascii_or_hyphen),
-             |d: &[u8]| String::from_utf8(d.to_vec()))
+             |d: &[u8]|
+                 match d.len() {
+                     0 => Err("Expected 1 or more characters"),
```

Comment:
I swear I tried this and it didn't work, ha! Must be something else in combination